File: src/types/transaction.ts
export const TransactionTypes = {
  INCOME: 'income',
  EXPENSE: 'expense',
} as const
export type TransactionType = typeof TransactionTypes[keyof typeof TransactionTypes]

export type Transaction = {
  id: number
  user_id: number
  title: string
  amount: number
  category: string
  type: TransactionType
  date: string        // ISO日付形式: yyyy-mm-dd
  created_at: string  // ISO datetime
  updated_at: string  // ISO datetime
}

File: app/models/transaction.py
from app.common.enums import TransactionType
from app.core.db import Base
from sqlalchemy import (
    Column,
    ForeignKey,
    Integer,
    Float,
    String,
    Date,
    DateTime,
    Enum as SAEnum,
)
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func


class Transaction(Base):
    __tablename__ = "transactions"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(String, nullable=False, index=True)
    memo = Column(String, nullable=True)
    type = Column(
        SAEnum(TransactionType), nullable=False, default=TransactionType.expense.value
    )
    amount = Column(Float, nullable=False)
    category_id = Column(
        Integer, ForeignKey("categories.id"), nullable=False, index=True
    )
    date = Column(Date, nullable=False)
    created_at = Column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

    category = relationship("Category", back_populates="transactions")
